### ISSUE TITLE
da broken (results in invalid 430 byte webm file)?

### ISSUE THREAD

**Issue by lol768** [mps-youtube/mps-youtube#513]
**Steps to reproduce:**

1. Launch mpsyt
2. Search for `cxGYHnTRMAw`
3. `da 1`
4. Observe the stream download URL is similar to `https://r5---sn-aigllnss.googlevideo.com/videoplayback?nh=IgpwcjA1LmxocjI1KgkxMjcuMC4wLjE&gir=yes&requiressl=yes&source=youtube&gcr=gb&dur=209.241&initcwndbps=2585000&pl=24&itag=251&ipbits=0&lmt=1449585933849749&sver=3&expire=1469389604&mime=audio%2Fwebm&key=yt6&upn=brifIlMqRF4&mm=31&mn=sn-aigllnss&mt=1469367312&mv=m&ms=au&fexp=9416891%2C9419452%2C9422596%2C9428398%2C9431012%2C9432564%2C9433092%2C9433096%2C9433221%2C9433424%2C9433946%2C9435524%2C9435526%2C9435876%2C9436607%2C9436834%2C9437066%2C9437085%2C9437552%2C9437742%2C9438140%2C9438527%2C9438663%2C9438871%2C9439327%2C9439580%2C9439652%2C9439688%2C9439882%2C9440545%2C9440881%2C9441371%2C9442424&sparams=clen%2Cdur%2Cei%2Cgcr%2Cgir%2Cid%2Cinitcwndbps%2Cip%2Cipbits%2Citag%2Ckeepalive%2Clmt%2Cmime%2Cmm%2Cmn%2Cms%2Cmv%2Cnh%2Cpl%2Crequiressl%2Csource%2Cupn%2Cexpire&ei=xMaUV_yuGYnPWfT_pZAB&ip=xxx.xxx.xxx.xxx&clen=3329073&id=o-AEaToQWep3lvHLahVIuK7PQ8xniylwzl19ziEniLelwo&keepalive=yes&signature=7B1D69815955DDD28DB39D4D32066B50DF2F6D31.1EA963C3D4DE2BA6C07A8EE5FE0ABD50B14F7C42&ratebypass=yes`
6. Observe the download finishes successfully.
7. `ls -lha` the downloaded file, it is 430 bytes:
```
-rw-rw-r-- 1 adam adam 430 Jul 24 14:58 Vixen - Cryin'.webm
``` 
Attempt to play the file with mpv:

```
➜  music mpv Vixen\ -\ Cryin\'.webm 
mpv: Symbol `av_pix_fmt_descriptors' has different size in shared object, consider re-linking
Playing: Vixen - Cryin'.webm
[mkv] Refusing to read element over 100 MB in size
Detected file format: Matroska
Clip info:
 TITLE: Cryin'
Video: no video
Audio: no audio
No video or audio streams selected.
```

**Versions:**

```
➜  mpsytv3 git:(develop)  git show HEAD | grep commit
commit 8508c46dae26ee04ea0f7ddcbb1bc0feb69add2a
➜  mpsytv3 git:(develop) ✗ sudo pip3 show pafy
---
Name: Pafy
Version: 0.3.72
Location: /usr/local/lib/python3.4/dist-packages
Requires: 
➜  mpsytv3 git:(develop) ✗ sudo pip3 show youtube-dl
---
Name: youtube-dl
Version: 2016.7.24
Location: /usr/local/lib/python3.4/dist-packages
Requires: 
```

**Other notes:**

* Downloading the 160k webm stream with `d 1` works.
* Hexdump of invalid file: https://gist.github.com/8ac13d24bed8e0439f4560f482e88858

**Comment by ids1024** [comment]
It seems to download correctly for me...

**Comment by lol768** [comment]
Yeah, it downloads correctly for a friend on IRC too though I can still reproduce it consistently on my machine - even on a different network. Very odd, not sure what'd cause it.

**Comment by ids1024** [comment]
I really don't know what would cause this. Do you always get an identical binary? Could you post it encoded in base64 (or the original binary). I tried using xxd to convert the hexdump back into a binary, but I'm not sure it worked properly.

**Comment by lol768** [comment]
>  Could you post it encoded in base64 (or the original binary)

It would appear so, yeah.

```
GkXfowEAAAAAAAAfQoaBAUL3gQFC8oEEQvOBCEKChHdlYm1Ch4ECQoWBAhhTgGcB/////////+wB
AAAAAAAA1gAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAVSalmAQAA
AAAAAE4q17GDD0JAe6mGQ3J5aW4nTYCNTGF2ZjU0LjYzLjEwNFdBjUxhdmY1NC42My4xMDRzpJBS
42vCfJqkWaECfUwz37EZ7AEAAAAAAAACAAAWVK5rAf////////+uAf/////////XgQFzxYEBnIEA
IrWcg2VuZ4iBAYaTQV9PUFVTL0VYUEVSSU1FTlRBTA==
```

Download: https://lol768.com/i/Vixen%20-%20Cryin'.webm

**Comment by ids1024** [comment]
I'm not sure what is happening, but I found something.

Here is the `strings` output of that file:
```
webmB
Cryin'M
Lavf54.63.104WA
Lavf54.63.104s
A_OPUS/EXPERIMENTAL
```

And this is it for the (longer, correct) file I get:
```
webmB
Cryin'M
Lavf57.41.100WA
Lavf57.41.100s
x`k3
A_OPUSV
```

I can't even say for sure if this is related to the problem, nor do I have any idea what to do about it but... it's something.

I don't know exactly what `A_OPUS/EXPERIMENTAL` means. Perhaps it is relevant.

**Comment by lol768** [comment]
Interesting find! I did some searching online of  `A_OPUS/EXPERIMENTAL` and found an ffmpeg patch related to it. I took another look at this today and saw it was executing this command to do the remuxing:

`/usr/bin/ffmpeg -y -i /media/adam/ssd/music/Vixen - Cryin'.webm.89290 -acodec copy -metadata title=Cryin' -metadata artist=Vixen -vn /media/adam/ssd/music/Vixen - Cryin'.webm`

So next, I checked the ffmpeg version:

```
$ ffmpeg -h
ffmpeg version 1.2.6-7:1.2.6-1~trusty1 Copyright (c) 2000-2014 the FFmpeg developers
  built on Apr 26 2014 18:52:58 with gcc 4.8 (Ubuntu 4.8.2-19ubuntu1)
```

This seemed old, so I did an `apt install ffmpeg` after adding a PPA and then redid all of the original steps.

```
$ ffmpeg -v
ffmpeg version N-80901-gfebc862 Copyright (c) 2000-2016 the FFmpeg developers
  built with gcc 4.8 (Ubuntu 4.8.4-2ubuntu1~14.04.3)
```

This gave me a 3.2M file that I could successfully play with `mpv`, so it would appear this is a (fixed) ffmpeg bug present in Ubuntu Trusty's repository? In any case, it seems fixed now.

Thanks a bunch for looking into this for me.
